src/dashboard_helper.js

```diff
@@ -59,11 +58,71 @@ function returnCurrentDate() {
   return today;
 }
 
-function appendToSection(paragraph) {
+function isTravelerPage() {
+  if (document.getElementById("traveler-page")) {
+    return true;
+  }
+}
+
+function appendToTravelerPage(paragraph) {
+  document.getElementById("traveler-page").appendChild(paragraph);
+}
+
+function isRequestedOrUpcomingTripsSection() {
   if (document.getElementsByTagName("SECTION").length === 1) {
-    document.getElementsByTagName("SECTION")[0].appendChild(paragraph);
-  } else {
-    document.getElementsByTagName("SECTION")[1].appendChild(paragraph);
+    return true;
+  }
+}
+
+function appendToRequestedOrUpcomingTripsSection(paragraph) {
+  document.getElementsByTagName("SECTION")[0].appendChild(paragraph);
+}
+
+function noSectionsHaveLoaded() {
+  if (document.getElementsByTagName("SECTION").length < 1) {
+    return true;
+  }
+}
+
+function appendToBody(paragraph) {
+  document.querySelector("body").appendChild(paragraph);
+}
+
+function isTodaysTripsSection() {
+  if (document.querySelectorAll("section")[1].innerText === "Today's Trips:") {
+    return true;
+  }
+}
+
+function appendToTodaysTripSection(paragraph) {
+  let todaysTripsSection = document.querySelectorAll("section")[1];
+  todaysTripsSection.setAttribute("id", "todays-trips");
+  document.getElementById("todays-trips").appendChild(paragraph);
+}
+
+function isPastTripsSection() {
+  if (
+    document.querySelectorAll("section")[1].innerText.includes("Past Trips")
+  ) {
+    return true;
+  }
+}
+
+function appendToPastTripsSection(paragraph) {
+  document.querySelectorAll("section")[1].appendChild(paragraph);
+}
+
+function appendToSection(paragraph) {
+  if (isTravelerPage()) {
+    appendToTravelerPage(paragraph);
+  } else if (isRequestedOrUpcomingTripsSection()) {
+    appendToRequestedOrUpcomingTripsSection(paragraph);
+  } else if (noSectionsHaveLoaded()) {
+    appendToBody(paragraph);
+  } else if (isTodaysTripsSection()) {
+    appendToTodaysTripSection(paragraph);
+  } else if (isPastTripsSection()) {
+    appendToPastTripsSection(paragraph);
```

Comment:
this conditional chain is much more readable. Still kind of crazy, but at least you can understand it !